homeassistant/components/lightmap/lightmap.py

```diff
@@ -2,4 +2,248 @@
+        stops.
+        """
+        radial_gradient = Element("radialGradient")
+        radial_gradient.set("id", f"radial-{self._sensor_svg_id}")
+        radial_gradient.set("cx", str(sensor_x + x_translation))
+        radial_gradient.set("cy", str(sensor_y + y_translation))
+        radial_gradient.set("fx", str(sensor_x + x_translation))
+        radial_gradient.set("fy", str(sensor_y + y_translation))
+        radial_gradient.set("r", str(radius_val))
+        radial_gradient.set("gradientTransfrom", "scale(1,1)")
+        radial_gradient.set("gradientUnits", "userSpaceOnUse")
+
+        first_gradient_stop = SubElement(radial_gradient, "stop")
+        first_gradient_stop.set("offset", "0%")
+        first_gradient_stop.set(
+            "style",
+            f"stop-color:rgba(255,255,10,1);stop-opacity:{str(radial_firststop_opacity)}",
+        )
+        first_gradient_stop.set("id", f"{radial_gradient.attrib.get("id")}-stop-1")
+
+        second_gradient_stop = SubElement(radial_gradient, "stop")
+        second_gradient_stop.set("offset", "100%")
+        second_gradient_stop.set(
+            "style", "stop-color:rgba(255,255,255,0); stop-opacity:0"
+        )
+        second_gradient_stop.set("id", f"{radial_gradient.attrib.get("id")}-stop-2")
+
+        return radial_gradient
+
+    def _create_overlapping_rectangle(self, room_rectangle_dimensions: dict) -> Element:
+        """
+        Creates a new rectangle that will
+        overlap the room and contain the radial gradient
+        """
+        rect_params = {
+            **room_rectangle_dimensions,
+            "style": f"fill:url(#radial-{self._sensor_svg_id});fill-opacity:1;fill-rule:nonzero;stroke:#000000;stroke-width:4.17796;stroke-dasharray:none;stroke-opacity:0;paint-order:stroke markers fill",
+        }
+
+        new_rectangle = Element("rect")
+
+        for key, value in rect_params.items():
+            new_rectangle.set(key, str(value))
+
+        return new_rectangle
+
+    def _add_lightmap_to_svg(
+        self,
+        room_parent_element: Element,
+        radial_gradient: Element,
+        overlapping_rect: Element,
+    ):
+        """
+        Adds the lightmap adjustments to the svg
+        """
+        room_element_parent = room_parent_element.getparent()
+        room_index = room_element_parent.index(room_parent_element)
+        # Put new rectangle on top of old rectangle.
+        room_element_parent.insert(room_index, overlapping_rect)
+        tree = room_element_parent.getroottree()
+        root = tree.getroot()
+        root.append(radial_gradient)
+        tree.write(self._svg_path)
+
+    def calculate_light_radial(self):
+        """
+        Calculate light radial distance.
+
+        Assumption: We assume that photoresistors are used where
+        the light intesnity is proportional to the resistance -
+        in other words, as light intensity increase, resistance increases.
```

Comment:
Yes but, in practice the photoresistor when the luminosity is higher the resistance is lower, and when the luminosity is low the resistance is higher. But yeah the value that we read is the luminosity that we read, high value, high luminosity low value, low luminosity.

It was just a technicality in the description, the code is great.
https://en.wikipedia.org/wiki/Photoresistor
https://www.sciencedirect.com/topics/engineering/photoresistors

Reply:
So I just don't understand this step here, we are reading high value when high luminosity, and low value when low luminosity right?

Reply:
All ist gud